File: leetcode_py/p1253.py
from typing import List


class Solution:
    def reconstructMatrix(self, upper: int, lower: int, colsum: List[int]) -> List[List[int]]:

        # 贪心，先看全为2和全为 0的，其他就是分配1 和 0
        n = len(colsum)
        res = [[0] * n for _ in range(2)]
        for i,v in enumerate(colsum):
            if upper < 0 or lower < 0:
                return []
            if v == 2:
                res[0][i] = res[1][i] = 1
                upper -= 1
                lower -= 1
            elif v == 1:
                if upper >lower:
                    upper -= 1
                    res[0][i] = 1
                else:
                    lower -= 1
                    res[1][i] = 1
        return res if upper == 0 and lower == 0 else []
            
       


if __name__ == '__main__':

        # 剩下的就是1 和 0的

File: leetcode_C++/p1253.cpp
/*
@Author: mrxhar
@Date:   2023-09-08 10:42:18
@Email:  1337904145@qq.com
@Blog:   https://leetcode.cn/problems/reconstruct-a-2-row-binary-matrix/description/?envType=daily-question&envId=2023-09-08
*/
#include <vector>
using std::vector;

class Solution {
public:
    vector<vector<int>> reconstructMatrix(int upper, int lower, vector<int>& colsum) {
        vector<vector<int>> ans(2, vector<int>(colsum.size(), 0));
        int n = colsum.size();
        for (int i = 0; i < n; i++) {
            if (upper < 0 || lower < 0) {
                return vector<vector<int>>();
            }
            if (colsum[i] == 2) {
                ans[0][i] = 1;
                ans[1][i] = 1;
                upper--;
                lower--;
            } else if (colsum[i] == 1) {
                if (upper > lower) {
                    ans[0][i] = 1;
                    upper--;
                } else {
                    ans[1][i] = 1;
                    lower--;
                }
            }
        }
        return (lower == 0 && upper == 0) ? ans : vector<vector<int>>();
    }
};

int main() {
    return 0;
}
